Userland/Libraries/LibSoftGPU/Device.cpp

```diff
@@ -688,10 +688,9 @@ void Device::draw_primitives(PrimitiveType primitive_type, FloatMatrix4x4 const&
                     } else if ((vertex.eye_coordinates.w() == 0.0f) && (light.position.w() != 0.0f)) {
                         vertex_to_light = -vertex.eye_coordinates;
                     } else {
-                        // FIXME: The OpenGL 1.5 spec says (in typical SGI fashion) _nothing_ about if P1 and P2 have a w value of 1, which would
-                        // then mean the light position has an implicit value of (0, 0, 0, 0). This doesn't make any logical sense, and it most likely
-                        // a typographical error. Most other GL implementations seem to just fix it to the distance from the vertex to the light, which
-                        // seems to work just fine.
+                        // FIXME: The OpenGL 1.5 spec says nothing about if P1 and P2 have a w value of 1, which would then mean the light position has an
```

Comment:
Since this comment was added in the first commit, lets squash any fixes to it (they're currently in the 4th commit)

Reply:
Whoops, I'll remove this diff. I accidentally did a fixup to remove the SGI reference but thought it was done in this commit.  Commits 3 and 4 should still be separate because they do different things (the whole lighting equation is changed by commit 3) and specular is implemented in commit 4